chess/board.h

```diff
@@ -26,21 +33,31 @@ namespace space {
 		Move(int v_sourceRank, int v_sourceFile, int v_destinationRank, int v_destinationFile, PieceType v_promotedPiece = PieceType::None) :
 			sourceRank(v_sourceRank), sourceFile(v_sourceFile), destinationRank(v_destinationRank), destinationFile(v_destinationFile), promotedPiece(v_promotedPiece)
 		{}
-		inline std::string as_string() const {
+		Move(const std::string &s);
```

Comment:
I am thinking of converting some of these types to "Plain old data" types (https://en.cppreference.com/w/cpp/named_req/PODType), so at some point i will convert these constructors to static factory functions.
Just FYI... for now it's perfectly fine.